lib/gemstash/http_client.rb

```diff
@@ -26,10 +26,11 @@ class HTTPClient
 
     DEFAULT_USER_AGENT = "Gemstash/#{Gemstash::VERSION}".freeze
 
-    def self.for(upstream)
+    def self.for(upstream, timeout = 20)
       client = Faraday.new(upstream.to_s) do |config|
         config.use FaradayMiddleware::FollowRedirects
         config.adapter :net_http
+        config.options.timeout = timeout
```

Comment:
How about fetch the config timeout here instead of passing it in? So `config.options.timeout = Gemstash::Env.current.config[:fetch_timeout]`. You could also use the helper, so in `Gemstash::HTTPClient` it could `extend Gemstash::Env::Helper` and then `Gemstash::HTTPClient.for` could do `config.options.timeout = gemstash_env.config[:fetch_timeout]`